File: accidentes/FicheroAccidentes-Java-IvanRC/src/main/java/mapper/lesividad/LesividadMapper.java
package mapper.lesividad;


import dto.lesividad.LesividadDto;
import model.Lesividad;
import utils.Utils;

public class LesividadMapper {
    public static LesividadDto toDto(Lesividad distrito){
        return new LesividadDto(
                Utils.getValorNumericoToString(distrito.getId()),
                distrito.getLesividad()
        );
    }

    public static Lesividad toLesividad(LesividadDto distritoDto){
        return new Lesividad(
                Utils.valorNumericoToInt(distritoDto.getId()),
                distritoDto.getLesividad()
        );
    }
}

File: accidentes/FicheroAccidentes-Kotlin-IvanRC/src/main/kotlin/mapper/lesividad/LesividadMapper.kt
package mapper.lesividad

import dto.lesividad.LesividadDto
import model.Lesividad

fun Lesividad.toDto(): LesividadDto{
    return LesividadDto(
        id = if(this.id != null ) this.id.toString() else "NULL",
        lesividad = if(this.lesividad != null) this.lesividad else "NULL"
    )
}

fun LesividadDto.toLesividad(): Lesividad{
    return Lesividad(
        id = if(this.id.matches(Regex("[0-9]+"))) this.id.toInt() else null,
        lesividad = if(this.lesividad != "NULL" ) this.lesividad else null
    )
}